glean-core/python/glean/_process_dispatcher.py

```diff
@@ -75,25 +69,27 @@ class ProcessDispatcher:
 
     # Store the last run process object so we can `join` it when starting
     # another process.
-    _last_process = None  # type: Optional[multiprocessing.Process]
+    _last_process = None  # type: Optional[subprocess.Popen]
 
     @classmethod
-    def dispatch(cls, func, args) -> Union[_SyncWorkWrapper, "multiprocessing.Process"]:
+    def dispatch(cls, func, args) -> Union[_SyncWorkWrapper, subprocess.Popen]:
         from . import Glean
 
         if Glean._configuration._allow_multiprocessing:
-            # Only import the multiprocessing library if it's actually needed
-            import multiprocessing  # noqa
-
             # We only want one of these processes running at a time, so if
             # there's already one, join on it. Therefore, this should not be
             # run from the main user thread.
             if cls._last_process is not None:
-                cls._last_process.join()
+                cls._last_process.wait()
                 cls._last_process = None
 
-            p = multiprocessing.Process(target=_work_wrapper, args=(func, args))
-            p.start()
+            p = subprocess.Popen(
+                [
+                    sys.executable,
+                    _process_dispatcher_helper.__file__,
+                    base64.b64encode(pickle.dumps((func, args))).decode("ascii"),
```

Comment:
Are there any limitations with sending base64-encoded and pickled data over?
I know there are arg length limit on different platforms, do we need to take care of that?

I know we probably control all code paths calling this, but we should still document the limitations (if any) for our future selfs.